spatialmath/capsule.go

```diff
@@ -373,3 +374,140 @@ func separatingAxisTest1D(positionDelta, capVec *r3.Vector, plane r3.Vector, hal
 	sum -= math.Abs(capVec.Dot(plane))
 	return sum
 }
+
+// CapsuleIntersectionWithPlane calculates the intersection of a geometry with a plane and returns
+// a list of points along the surface of the geometry at the points of intersection.
+// It returns an error if the geometry type is unsupported or if points cannot be computed.
+// The points returned are in order, in frame of the capsule's parent, and follow the right hand rule around the plane normal.
+func CapsuleIntersectionWithPlane(g Geometry, planeNormal, planePoint r3.Vector, numPoints int) ([]r3.Vector, error) {
```

Comment:
I can follow the comments but its really hard for me to reveiw the content of the math without rederiving what you have done.  I can do this, but before I do so I want to check in with you to see how important reviewing this section feels to the review

Reply:
I think not so much. It works, clearly, based on the demo. I don't think this is such a crucial function that it needs in-depth examination. If something were massively wrong I expect I would have encountered and fixed it during the demo (as indeed I did with several bugs).